operator/controller/src/main/java/io/apicurio/registry/operator/resource/ResourceFactory.java

```diff
@@ -28,16 +38,157 @@ public class ResourceFactory {
     public static final String UI_CONTAINER_NAME = "apicurio-registry-ui";
 
     public Deployment getDefaultAppDeployment(ApicurioRegistry3 primary) {
-        var r = getDefaultResource(primary, Deployment.class, RESOURCE_TYPE_DEPLOYMENT, COMPONENT_APP);
-        addDefaultLabels(r.getSpec().getTemplate().getMetadata().getLabels(), primary, COMPONENT_APP);
+        var r = new Deployment();
+
+        // Metadata
+        if (r.getMetadata() == null) {
```

Comment:
are those null checks necessary?
we are building a brand new `Deployment` object every time, they are going to be always `null`

Reply:
the comment applies to all the occurrences.

Reply:
Fai point, but it applies only for occurrences outside the PTS.

Reply:
right, when we split the merging logic from the "default" generation everything should flow nicely.

Reply:
Done.